Only first page of labels is read, resulting in failure when creating one that exists

I tried running Maven core migration, but it failed right at the start when creating labels.

I think the handling of the response is not correct


> 16:18:38.162 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "[\r][\n]"
16:18:38.163 [main] INFO  io.pivotal.migration.MigrationClient - **Existing labels: [backport, backport-to-4.0.x, breaking, bug, build, chore, dependencies, deprecated, documentation, duplicate, enhancement, forward-port-to-master, good first issue, hacktoberfest-accepted, help wanted, in-progress, invalid, java, maintenance, mvn3, mvn40, not-for-merge, performance, priority:blocker, priority:critical, priority:major, priority:minor, priority:trivial, removed, skip-changelog]**
16:18:38.164 [main] INFO  io.pivotal.migration.MigrationClient - Creating labels: [{name=waiting-for-feedback, color=e3d9fc}]
16:18:38.164 [main] DEBUG io.pivotal.migration.MigrationClient - Creating label: "{name=waiting-for-feedback, color=e3d9fc}"

The "existing labels lists only contains those of the first page

https://github.com/apache/maven/labels?page=1

But Maven has more labels as listed there on secnd page, including the "waiting-for-feedback" one, which then failes the migration run.


```
16:18:37.914 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "Connection: Keep-Alive[\r][\n]"
16:18:37.916 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "User-Agent: Apache-HttpClient/4.5.14 (Java/17.0.6)[\r][\n]"
16:18:37.917 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "Accept-Encoding: gzip,deflate[\r][\n]"
16:18:37.917 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "[\r][\n]"
16:18:38.117 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "HTTP/1.1 200 OK[\r][\n]"
16:18:38.118 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Date: Fri, 04 Jul 2025 14:18:38 GMT[\r][\n]"
16:18:38.120 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Content-Type: application/json; charset=utf-8[\r][\n]"
16:18:38.121 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Cache-Control: private, max-age=60, s-maxage=60[\r][\n]"
16:18:38.121 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Vary: Accept, Authorization, Cookie, X-GitHub-OTP,Accept-Encoding, Accept, X-Requested-With[\r][\n]"
16:18:38.122 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "ETag: W/"cb58918cc2de21638421e82dde58be85629d50d7b60b5a21a8708ba272c445b9"[\r][\n]"
16:18:38.125 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-OAuth-Scopes: public_repo, repo:status[\r][\n]"
16:18:38.125 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-Accepted-OAuth-Scopes: repo[\r][\n]"
16:18:38.126 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "github-authentication-token-expiration: 2025-08-06 07:58:40 UTC[\r][\n]"
16:18:38.126 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-GitHub-Media-Type: github.v3[\r][\n]"
16:18:38.127 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Link: <https://api.github.com/repositories/206483/labels?page=2>; rel="next", <https://api.github.com/repositories/206483/labels?page=2>; rel="last"[\r][\n]"
16:18:38.127 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "x-github-api-version-selected: 2022-11-28[\r][\n]"
16:18:38.128 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-RateLimit-Limit: 5000[\r][\n]"
16:18:38.128 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-RateLimit-Remaining: 4909[\r][\n]"
16:18:38.128 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-RateLimit-Reset: 1751642111[\r][\n]"
16:18:38.129 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-RateLimit-Used: 91[\r][\n]"
16:18:38.129 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-RateLimit-Resource: core[\r][\n]"
16:18:38.130 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset[\r][\n]"
16:18:38.133 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Access-Control-Allow-Origin: *[\r][\n]"
16:18:38.134 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Strict-Transport-Security: max-age=31536000; includeSubdomains; preload[\r][\n]"
16:18:38.134 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-Frame-Options: deny[\r][\n]"
16:18:38.135 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-Content-Type-Options: nosniff[\r][\n]"
16:18:38.135 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-XSS-Protection: 0[\r][\n]"
16:18:38.142 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Referrer-Policy: origin-when-cross-origin, strict-origi"
16:18:38.142 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "n-when-cross-origin[\r][\n]"
16:18:38.143 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Content-Security-Policy: default-src 'none'[\r][\n]"
16:18:38.143 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Content-Encoding: gzip[\r][\n]"
16:18:38.144 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Transfer-Encoding: chunked[\r][\n]"
16:18:38.144 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Server: github.com[\r][\n]"
16:18:38.145 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-GitHub-Request-Id: 2FF7:38E22:1910F55:1995EDC:6867E2BE[\r][\n]"
16:18:38.145 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "[\r][\n]"
16:18:38.146 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "65F[\r][\n]"
16:18:38.146 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "[0x1f][0x8b][0x8][0x0][0x0][0x0][0x0][0x0][0x0][0x3][0xad]XkS[0xe3]6[0x14][0xfd]+[0x9a][0xcc]t[0xfa][0x85],~[0xca]6[0xdf][0x92][0x90][0x5]Z[0x2][0xec][0xa3][0xb3][0xdd]v:;[0x8a]-'"[0xb6][0xe5][0xca]r[0x2][0xed][0xec]⌂[0xef][0x95][0x13]?B[0x2]bq[0xcd]0[0x8c][0xaf][0x82][0xa2][0xa3]{[0xee][0xb9][0x8f]?[0xff][0x1d][0xb0]hp[0xe6][0xfb]8p[0x2][0x8c]-[0xfb]d[0x90][0xf1][0x88]~S[0xc6][0xc1][0xf5][0xe8][0xdb]js~;[0x1a][0xcd][0xd2][0x95][0xef][0x8f][0xe0][0x99]L⌂[0xcd].[0xaf]7[0x83][0x93]A)[0x12][0xf8][0xc4]R[0xca][0xbc]8;=%9{[0xb7]`rY[0xce][0xdf][0x85]<=[0x15]4[0xe7][0x5][0x18]I[0xb8][0xa4][0xa7])Y[0xd3][0xec]4!s[0x9a][0x14][0xa7]s[0x12][0xae]r.$[0xec][0x90][0x91][0x94][0xc2][0x16][0x1d]K[0xc8][0x13].[0xc0][0x14]V[0xf]|$[0xa2]1)[0x13]98[0x8b]IRP[0xf5]^[0x84][0x82][0xe5][0x92][0xf1][0xc]>6[0xf8]~[0xd2]9[0xbd]o[0x18][0x9e][0xfe][0xf4][0xe9][0x5][0xfb][0xd0][0xff][0xf4]C[0xc9][0x87][0xce];[0xe3][0xdd][0xc3]![0x8c][0xee]R[0x8d]'r=/r~[0x0][0x8f]gc[0xc3][0x5]8[0x9e][0xc6][0x1b][0x93][0xeb]/[0xb7][0xb3]>x[0x4]%+[0x96]-:0ZK}z[0xea]P[0x8c][0x3][0xed][0xe9][0xef][0xca]$A[0x82][0xfe]][0xd2]B[0x16]H.[0x89]Ds[0xb5][0x19][0xa2][0xf][0xac][0x90][0xf0]%([0xa6]D[0x96][0x82][0x16][0x8d][0xdf]\[0xcb]t-[0xc0][0xea][0xb8]/[0xe3][0x1c][0xdf]H1[0x11]}p[0x96]][0x88][0xd5]K[0xe3][0x1b][0xcf]&[0xe][0xe9][0xa2][0x93][0xa2]<[0xa0][0xda]'[0x9e]R[0xb9]T X[0x91][0xfd],[0xd1][0x86][0x8b][0xea][0xde][0x1a][0x6][0xd6][0x1e][0xd3]1Py[0xec][0xba][0x17][0x12][0x96]D[0x1d]w[0x95][0xdb][0xd7][0x1a]M`ZS[0xeb]M[0xbe][\n]"
16:18:38.147 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "[0x97]$[P[0xf0][0x1b]Es[0xb5])[0xca][0x5][0xf]i[0xd1]:[0xcb][0xb6][0xc][0x13][0x9b]~[0x80][0xfd][0xae][0xb3]f[0xe7][0xbf][0xe1][0xcf][0x17][0xef][0xd9][0x1f]_6[0xff][0xcc][0xee]G[0xf]7[0xf7]S[0xe7][0xf6][0xf3]WE[0xf6]7*D[0xb8][0xe4][0x82][0xb6][0x8][0xeb][0xd7][0x1a][0xa1]gO[0xde][0xfb]X[0xcb][0xc6][0xc6]3[0x8e]e[0x1a]Xi[0x3]~[0x96]c[0x1e][0x1e][0xd2][0xe9][0xec][0xed]'[0x8e]hN[0xb3][0x88]f![0x3]n7[0xba][0xf6][0xc4]Z[0x9f][0xdf][0xb0]1[0x8e][0xf4][0xe7]?[0x12]Me[0x1e][0x11]I[0x11]A[0xcd][0xce][0x8f](f[0x9]=$[0xe1][0xf3]P[0xb7]"^[0xc9][0xc6][0xa8][0x17]`AC8L[0x87][0x89]p[0xa8][0xd6]V[0x83][0x9d]x[0x86]i[Zg[0x1d][0x1][0xdb]lw[0xa8][0x1a]^[0xe0][0xc2][0x8f][0xe3][0xfb][0x1a]u[0x1c][[0xa9]}[0xb9][0xec][0x91][0xab]"[0x1e][0x96])[0xcd]$[0xa9][0xf2]M[0xeb][0xd8]'[0xe6][0xc6][0xb3][0x86][0xe7][0x86]Z%[0xb9]J![0xb0][0xd6]T[0xed][ .[0x10][0x89]"[0xa6][0xf6][0x7][0xc9][0xe4]h[0xff][0x1b]kqi[0x0][[0xcf]R[0xb8][0xf2]k[0x5]x[0xa1][0xc4][0xed][0x8d][0xa1][0x17][0x95]y[0xc2][0x94]_;`;[0xa6][0x1a]h[0x18]Gv[0xe4]u[0xbd]zL2?/Y[0x1]jY[0x94]T[0xc1][0xcc];[0xd9][0x1][0x91][0x4][0x12]C[0xf4][0xb8][0xcd][0xc][0xad][0xc2]x[0xbe][0x8b]-[0xd3]q[0xb1][0xa9][0xc1]iN?~[0xba][0xea]![0xa2]4[0x3][0xb1][0xb]+[0x1f][0xb4]H[0xf7][0x8d]5VbQJc[0x1d][0xd6][0x1b][0xba][0xa9]y[0xaa][0xc0][0xee][0xb2][0xe0][0x93][0xfa][0xc4]q[\r]]v[0x98][0xfe][0x9a][0xfe]n[0xf6]`l[0xcc][0xc5][0x86][0x88]h[0xa8]*,U[0x88][0xa4][0xa4][0x90]T[0xb4][0x18][0x9f]][0xaf][0xe1]b{N[0xc9][0x1e][0xdc][0x97]+[0xaf][0x9a][0x9a][0x81][0xce]e[0x8a][0x9a][0xac][0x7][0xb2][0x5][0xe7][0xd1]O[0x96][0x11]3QH[0xf8][[0xf1][0xaa][0xc5][0xa5]VA[0xb]am[0xcb]8Xi[0xf2][0x85][0xe1]z[0xb1][0xd6][0x81][0x17][0xd5][0x6][0xe0][0xba][0x8c]n[0xa0][0x80][0xa5][0xa2]e[0xa5][0xe5]8[0x81][0xe1][0x4][0x16][0xde][0x93][0x9b]N[0xde]{[0xbc]9[0xff][0xe0][0xce][0xd4][0xef][0xfd]W[0xfb][0xed][0xc1][0xb7][0x84][0xca]X[0xf2]9[0x15]1[0x14]PC[0x12][0x86]4[0xdf][0x13][0xd8]g[0xd7]k[0xa4][0xc1]xd[0xfc]Hflde/[0xa1][0x1f][0xd6][0xfc][0x95][0xac][0xb0][0x1e][0xe9]bI[0x93][0x1c]|[0xb6]![0xd9]> 0[0xa3][0xc6][0xd8][0xaa][0xa8][0x8f][0xbd][0xbd][0x94]qL\[0xa6][0xf]R[0x10]D[0xa4][0x84][0x10][0x6][0xf1][0x4][0xaf][0x83][0xe7]h[0x4][0x9][0xa9][0xd3][0x13]8[0x6][0xf6]|[0xd3]xYL&[0xd3][0xf7][0xf3][0x8d][0xd3]C4Y6[0x4]E_@][0xdb]I[0xfe][0xfb][0xc6]FL[0x88]z[0xba]br,[0xba]2[0x90][0xca][0x3][0xe9][0xd7]([0xc7]V[0xfa]{[0xc4][0x17][0xcb][0xd6]$[0x81][0xa6][0xaf][0xc9]r[0xad][0xa1]>[0xfd]a[0x1f]p[0xcc]3[0x95][0xec]G[0x9c]VerAi[0x8a][0x4][,e[0xe3][0x17][0x1c]`[0xdb][0xc4][0x86][0x11][0xe8]j[0xfe]l[0x93][0xac].{ [0xba]'ku[0xd5][0xbb].s[0xf7]Vc[0x89]cbY[0xa6][0xce][0x13][0x83]#[0x85][0xc9][0xae][\n]"
16:18:38.150 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "[0xfb][0x5]vG!t[0xca][\r][0xb0]&[0x9c][0x82][0x97][0x9][0xb7][\r][0xa7][0x1e][0xd9]+%[0xc][0x2])S[0x19][0xac][0xc5][0xb7]ol$[0xe1]u[0xed]@[0xeb][0x1c][0xdf]q[0xc][0xc3]5-[0x9d]$$C[0xdf][0x9b][0xf4][0x90][0x84]t[0x9d])[0xa9][0xdc]9g[0xf7]V[0x9f][0xda]t=⌂|[0xae]uN'[0xd4]=3[0x0][0x81][0xd6]%[0xa1][0xc9]t[0xf6]-[\n]"
16:18:38.151 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "{P[\n]"
16:18:38.154 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "[0xce][0xe9][0x18]{[0xc7][0xae]^[0xeb]sc[0xcb][0xb5]F[0xfe][0xeb][0xcf][0xed][0xd9][0x96]cB[0x85][0xec]j[0xea][0xf5][0xb1][0xbc][0xb2]m[0xd1][0xe3][0xb6]3.[0x87][0x90][0xdc][0x86][0x90][0xd7][0x16][0x1d][0xce]<5[0xd7]8[0xc6][0xe6]"
16:18:38.155 [main] DEBUG io.pivotal.migration.MigrationClient - 200 OK {X-RateLimit-Remaining:4909}
16:18:38.158 [main] DEBUG io.pivotal.github.GitHubRestTemplate - Response 200 OK
16:18:38.159 [main] DEBUG io.pivotal.github.GitHubRestTemplate - Reading to [java.util.List<java.util.Map<java.lang.String, java.lang.Object>>]
16:18:38.160 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "8[0xc0][\n]"
16:18:38.160 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "[0xd8][0xab][0xc7]/[0x0][0x2][0xdb]8[0xd0]J[0xed][0xf5][0x9d][0x81]{Hm[0xe][0xc9][0x91][0x8b]t[0x9f][0xf9][0xfb][0xc6][0x1a][0x83][0xef][0x5][0x11][0xfd][0x91][0x11][0x92]a[0x98][0x81][0xe5][0x9a][0x8e][0xae][0xa9][0xb0][0xef]n7[0x17]=8[0x94][0xb][0xc6][0x5][0x93][0x8f]g[0xf3][0x84][0x87][0xab]nqvd[0xa5]F[0x13][0x86][0x6]<Z[0x8f][0x8c][0xd5][0x96][0xa0][0xbe]tM[0x13][0x9e][0xab][0xf2][0x16][0x91],:[0x85][0xb2]FBM[0xa1][0xa6][0x17]jnq[0xa2]Z[0xfb][0xa8][0xc][0xab][0xc4][0x19][0xf2][0x12]z}`[0x2][0x12]e[0xd6]F[0xd4][0xee]6\K[0xa3]e[0xea]6[0xae]z0[0xb3][0xc1][0xc]c=[0x9][0xad]G[0xd2]F[0xd7][0xb1][0xa5]V[0xbe]_u[0x1f][0x13]A[0x8a]%-NP[0xc2][0xb]h[0xb6]b[0x4][\r]49[0x81][0xac][0xb4][0xa6][0x82][0xa2][0x94][0xa6]\<[0xa2][0x4]fT[0x7][0xc0][0x1d][0x9d][0x88]+[0xe0][0x97]=D[0xbc]A[0x97][0x92]{[0x98]y6Rx`[0xaf]!C[0xc6][0xc]^A[0x81][0x99][0xda][0xae][0xc1][0x1b][0x97]Y[0xe5][0xe6][0x3]|[0xae]&T[0xc7][0x15][0xbe]>[0xa1]Z[0xd3]<e[0xd9]q|;{[0x8b][0xf]CU[0xa0][0xa5][0xf8]L[0xfd][0xdb][0x1][0xbe][0x13][0xd5]uq[0x98]]A[0xa3])[0xf8]<[0x81][0xaa]c[0x3]/[0x14]QR<V[0xa4]'[0x82][0x97]Y[0xa4][0xea]D[0x18]S[0x14][0xaa][0xf1]k[\n]"
16:18:38.160 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "[0xc5][0x9a][0xeb][0xba][0x16][0xa6][0xba][0x92][0xff]#[0xf2][0xa5]`kv[0x94][0xea][0xed]J{-[0xb6][0xfd][0x8a]k[0x99][0xf0][0x2][0x6][0x94],l[0xe0]'l[0x5][0xc][0x87]6)[0xa7]IB#u[0x7]Q5o[0x0][0x1b][0x14]z[0x8b][0xc]l[0x92]>t[0xae][0xa1][0x9e]`jR[0xff][0xa4][0x1a][0x1e][0xf5]`[0x86][0x80][0xb0][w'G[0xad][0xa1]I@[0x13][0x18][0x5][0xb8]Z.[0x1c][0xa9][0xce][0xb6]{[0x1d][0xce][0x8c]|[0xb][0x9b][0xae][0xed]X[0xba][0x89][0xfa][0xd8][0xbb]0&}[0xe2][0xba]X[0xb1]|[0xb8][0x9d][0xa5]&\[0xdd][0xd2][0xae][0xc4]9[0xb0][0xd7]X[0xf][0xab][0xea]c=[0xc1][0xe0][0x13][0xec][0xb][0xc3]Y[0xe0][0xef][0xdd]G[0x14][0xb][0x9e]"RJ[0x9e][0xc2]h[\n]"
16:18:38.161 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "D3yD[0xcd]W[0xa2][0x5][0xcd][0xa8][0xd8][0x8e][0xac][0xbe][0xff][0xf5][0x1f][0x94][0xaf])[0xe1][0xdf][0x19][0x0][0x0][\r][\n]"
16:18:38.162 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "0[\r][\n]"
16:18:38.162 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "[\r][\n]"
16:18:38.163 [main] INFO  io.pivotal.migration.MigrationClient - Existing labels: [backport, backport-to-4.0.x, breaking, bug, build, chore, dependencies, deprecated, documentation, duplicate, enhancement, forward-port-to-master, good first issue, hacktoberfest-accepted, help wanted, in-progress, invalid, java, maintenance, mvn3, mvn40, not-for-merge, performance, priority:blocker, priority:critical, priority:major, priority:minor, priority:trivial, removed, skip-changelog]
16:18:38.164 [main] INFO  io.pivotal.migration.MigrationClient - Creating labels: [{name=waiting-for-feedback, color=e3d9fc}]
16:18:38.164 [main] DEBUG io.pivotal.migration.MigrationClient - Creating label: "{name=waiting-for-feedback, color=e3d9fc}"
16:18:40.182 [main] DEBUG io.pivotal.github.GitHubRestTemplate - HTTP POST https://api.github.com/repos/apache/maven/labels
16:18:40.182 [main] DEBUG io.pivotal.github.GitHubRestTemplate - Accept=[application/json, application/*+json]
16:18:40.184 [main] DEBUG io.pivotal.github.GitHubRestTemplate - Writing [{name=waiting-for-feedback, color=e3d9fc}] with org.springframework.http.converter.json.MappingJackson2HttpMessageConverter
16:18:40.185 [main] DEBUG io.pivotal.migration.MigrationClient - POST /repos/apache/maven/labels
16:18:40.186 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "[read] I/O error: Read timed out"
16:18:40.186 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "POST /repos/apache/maven/labels HTTP/1.1[\r][\n]"
16:18:40.187 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "Accept: application/json, application/*+json[\r][\n]"
16:18:40.187 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "Authorization: token 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[\r][\n]"
16:18:40.188 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "Content-Type: application/json[\r][\n]"
16:18:40.188 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "Content-Length: 48[\r][\n]"
16:18:40.189 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "Host: api.github.com[\r][\n]"
16:18:40.192 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "Connection: Keep-Alive[\r][\n]"
16:18:40.192 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "User-Agent: Apache-HttpClient/4.5.14 (Java/17.0.6)[\r][\n]"
16:18:40.193 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "Accept-Encoding: gzip,deflate[\r][\n]"
16:18:40.193 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "[\r][\n]"
16:18:40.194 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "{"name":"waiting-for-feedback","color":"e3d9fc"}"
16:18:40.382 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "HTTP/1.1 422 Unprocessable Entity[\r][\n]"
16:18:40.382 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Date: Fri, 04 Jul 2025 14:18:41 GMT[\r][\n]"
16:18:40.385 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Content-Type: application/json; charset=utf-8[\r][\n]"
16:18:40.385 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Content-Length: 197[\r][\n]"
16:18:40.386 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-OAuth-Scopes: public_repo, repo:status[\r][\n]"
16:18:40.386 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-Accepted-OAuth-Scopes: [\r][\n]"
16:18:40.387 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "github-authentication-token-expiration: 2025-08-06 07:58:40 UTC[\r][\n]"
16:18:40.388 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-GitHub-Media-Type: github.v3[\r][\n]"
16:18:40.388 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "x-github-api-version-selected: 2022-11-28[\r][\n]"
16:18:40.389 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-RateLimit-Limit: 5000[\r][\n]"
16:18:40.392 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-RateLimit-Remaining: 4908[\r][\n]"
16:18:40.392 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-RateLimit-Reset: 1751642111[\r][\n]"
16:18:40.393 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-RateLimit-Used: 92[\r][\n]"
16:18:40.393 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-RateLimit-Resource: core[\r][\n]"
16:18:40.394 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset[\r][\n]"
16:18:40.394 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Access-Control-Allow-Origin: *[\r][\n]"
16:18:40.395 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Strict-Transport-Security: max-age=31536000; includeSubdomains; preload[\r][\n]"
16:18:40.395 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-Frame-Options: deny[\r][\n]"
16:18:40.396 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-Content-Type-Options: nosniff[\r][\n]"
16:18:40.396 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-XSS-Protection: 0[\r][\n]"
16:18:40.397 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin[\r][\n]"
16:18:40.397 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Content-Security-Policy: default-src 'none'[\r][\n]"
16:18:40.400 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Vary: Accept-Encoding, Accept, X-Requested-With[\r][\n]"
16:18:40.404 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Server: github.com[\r][\n]"
16:18:40.404 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-GitHub-Request-Id: 2FF7:38E22:191194B:19968F2:6867E2BE[\r][\n]"
16:18:40.405 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "[\r][\n]"
16:18:40.408 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "{"message":"Validation Failed","errors":[{"resource":"Label","code":"already_exists","field":"name"}],"documentation_url":"https://docs.gi"
16:18:40.409 [main] DEBUG io.pivotal.migration.MigrationClient - 422 UNPROCESSABLE_ENTITY {X-RateLimit-Remaining:4908}
16:18:40.409 [main] DEBUG io.pivotal.github.GitHubRestTemplate - Response 422 UNPROCESSABLE_ENTITY
16:18:40.410 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "thub.com/rest/issues/labels#create-a-label","status":"422"}"
16:18:40.414 [main] INFO  o.s.b.a.l.ConditionEvaluationReportLoggingListener -

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
16:18:40.427 [main] ERROR o.s.boot.SpringApplication - Application run failed
java.lang.IllegalStateException: Failed to execute CommandLineRunner
        at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:771)
        at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:752)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:314)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1303)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1292)
        at io.pivotal.migration.MigrationApp.main(MigrationApp.java:67)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:108)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
        at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:65)
Caused by: org.springframework.web.client.HttpClientErrorException$UnprocessableEntity: 422 Unprocessable Entity: "{"message":"Validation Failed","errors":[{"resource":"Label","code":"already_exists","field":"name"}],"documentation_url":"https://docs.github.com/rest/issues/labels#create-a-label","status":"422"}"
        at org.springframework.web.client.HttpClientErrorException.create(HttpClientErrorException.java:141)
        at org.springframework.web.client.DefaultResponseErrorHandler.handleError(DefaultResponseErrorHandler.java:168)
        at org.springframework.web.client.DefaultResponseErrorHandler.handleError(DefaultResponseErrorHandler.java:122)
        at org.springframework.web.client.ResponseErrorHandler.handleError(ResponseErrorHandler.java:63)
        at org.springframework.web.client.RestTemplate.handleResponse(RestTemplate.java:825)
        at io.pivotal.github.GitHubRestTemplate.handleResponse(GitHubRestTemplate.java:121)
        at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:783)
        at io.pivotal.github.GitHubRestTemplate.doExecuteExtended(GitHubRestTemplate.java:75)
        at io.pivotal.github.GitHubRestTemplate.doExecute(GitHubRestTemplate.java:59)
        at org.springframework.web.client.RestTemplate.exchange(RestTemplate.java:676)
        at io.pivotal.migration.MigrationClient.createLabels(MigrationClient.java:261)
        at io.pivotal.migration.MigrationClient.createLabelsIfNotExist(MigrationClient.java:242)
        at io.pivotal.migration.MigrationApp.run(MigrationApp.java:138)
        at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:768)
        ... 13 common frames omitted
```

--- a/src/main/java/io/pivotal/migration/MigrationClient.java
+++ b/src/main/java/io/pivotal/migration/MigrationClient.java
@@ -243,7 +243,7 @@ public class  MigrationClient {
 	}
 
 	private List<String> findExistingLabels() {
-		RequestEntity<Void> requestEntity = getRepositoryRequestBuilder(HttpMethod.GET, "/labels").build();
+		RequestEntity<Void> requestEntity = getRepositoryRequestBuilder(HttpMethod.GET, "/labels?per_page=100").build();
 		List<Map<String, Object>> exitingsLabels = getRest().exchange(requestEntity, LIST_OF_MAPS_TYPE).getBody();
 		List<String> existingLabelNames = exitingsLabels.stream().map(labelObject -> (String) labelObject.get("name")).toList();
 		logger.info("Existing labels: {}", existingLabelNames);

